src/worker.py

```diff
@@ -110,21 +110,39 @@ def get_synced_msgs(self) -> Optional[Dict[str, BasePacket]]:
                 return sync_msgs
         return None
 
-    def get_most_recent_color_msg(self) -> Optional[BasePacket]:
-        return self._get_most_recent_msg("color")
+    def _add_msgs_from_queue(
+        self, frame_type: Literal["color", "depth"], queue_handler: QueuePacketHandler
+    ) -> None:
+        queue_obj = queue_handler.get_queue()
+        with queue_obj.mutex:
+            q_snapshot = list(queue_obj.queue)
 
-    def get_most_recent_depth_msg(self) -> Optional[BasePacket]:
-        return self._get_most_recent_msg("depth")
+        for msg in q_snapshot:
+            self.add_msg(msg, frame_type, msg.get_sequence_num())
 
     def _get_most_recent_msg(
-        self, frame_type: Literal["color", "depth"]
+        self, q_handler: QueuePacketHandler, frame_type: Literal["color", "depth"]
     ) -> Optional[BasePacket]:
+        self._add_msgs_from_queue(frame_type, q_handler)
+        while len(self.msgs) < 1:
```

Comment:
This is the key line here. To wait until there is at least one message before trying to find a message on the queue